services/slam/client_test.go

```diff
@@ -251,6 +292,13 @@ func TestClientFailingService(t *testing.T) {
 
 		failingSLAMClient := slam.NewClientFromConn(context.Background(), conn, slam.Named(nameFail).String(), logger)
 
+		// context cancel
```

Comment:
add context cancel check of client because streams will always produce a valid client otherwise due to the decoupling of the client connection and data request